my-account/app/myaccount/mydetails/MyPersonalDetailsPageController.java

```diff
@@ -0,0 +1,171 @@
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static myaccount.CustomerSessionUtils.overwriteCustomerSessionData;
+
+@Singleton
+public class MyPersonalDetailsPageController extends MyAccountController {
+
+    protected final ProductDataConfig productDataConfig;
+    protected final Form<MyPersonalDetailsFormData> myPersonalDetailsUnboundForm;
+
+    @Inject
+    public MyPersonalDetailsPageController(final ControllerDependency controllerDependency,
+                                           final ProductDataConfig productDataConfig, final FormFactory formFactory) {
+        super(controllerDependency);
+        this.productDataConfig = productDataConfig;
+        this.myPersonalDetailsUnboundForm = formFactory.form(MyPersonalDetailsFormData.class);
+    }
+
+    @AddCSRFToken
+    public CompletionStage<Result> show(final String languageTag) {
+        final UserContext userContext = userContext(languageTag);
+        return getCustomer(session())
+                .thenComposeAsync(customerOpt -> customerOpt
+                        .map(customer -> handleFoundCustomer(customer, userContext))
+                        .orElseGet(() -> handleNotFoundCustomer(userContext)),
+                        HttpExecution.defaultContext());
+    }
+
+    @RequireCSRFCheck
+    public CompletionStage<Result> process(final String languageTag) {
+        final UserContext userContext = userContext(languageTag);
+        final Form<MyPersonalDetailsFormData> myPersonalDetailsForm = myPersonalDetailsUnboundForm.bindFromRequest();
+        return getCustomer(session())
+                .thenComposeAsync(customerOpt -> customerOpt
+                        .map(customer -> {
+                            if (myPersonalDetailsForm.hasErrors()) {
+                                return handleFormErrors(myPersonalDetailsForm, customer, userContext);
+                            } else {
+                                final CompletionStage<Result> resultStage = updateCustomer(customer, myPersonalDetailsForm.get())
+                                        .thenComposeAsync(updatedCustomer -> handleSuccessfulCustomerUpdate(updatedCustomer, userContext), HttpExecution.defaultContext());
+                                return recoverWithAsync(resultStage, HttpExecution.defaultContext(), throwable ->
+                                        handleSetShippingToCartError(throwable, myPersonalDetailsForm, customer, userContext));
+                            }
+                        })
+                        .orElseGet(() -> handleNotFoundCustomer(userContext)),
+                        HttpExecution.defaultContext());
+    }
+
+    protected CompletionStage<Customer> updateCustomer(final Customer customer, final MyPersonalDetailsFormData newCustomerData) {
+        final List<UpdateAction<Customer>> updateActions = new ArrayList<>();
+        if (!Objects.equals(customer.getTitle(), newCustomerData.getTitle())) {
```

Comment:
Yes, I wanted to extract something from there too, especially since it's something that will happen often with other forms. Couldn't spend much time to find something better, though.